tests/plugin_filter/util.py

```
@@ -0,0 +1,218 @@
+def create_template(
+    tmp_db, name: str = "test_template", description: str = "test template"
+) -> str:
+    template_exists = tmp_db.session.query(
+        exists().where(UiTemplate.name == name)
+    ).scalar()
+    if template_exists:
+        return tmp_db.session.query(UiTemplate.id).first()[0]
+    template = UiTemplate(
+        name=name,
+        description=description,
+    )
+    tmp_db.session.add(template)
+    tmp_db.session.commit()
+    template_id = (
+        tmp_db.session.query(UiTemplate.id)
+        .filter(UiTemplate.name == template.name)
+        .first()[0]
+    )
+    return template_id
+
+
+def create_or_update_plugin(
+    tmp_db,
+    name: str = "test-plugin",
+    version: str = "0.0.0",
+    tags: list[PluginTag] | None = None,
+    description: str = "descr",
+) -> RAMP:
+    if not tags:
+        tags = []
+    plugin_exists = tmp_db.session.query(
+        exists().where(RAMP.name == name, RAMP.version == version)
+    ).scalar()
+    if plugin_exists:
+        plugin = (
+            tmp_db.session.query(RAMP)
+            .filter(RAMP.name == name, RAMP.version == version)
+            .first()
+        )
+        plugin.plugin_id = f"{name}@{version}"
+        plugin.name = name
+        plugin.version = version
+        plugin.tags = tags
+        plugin.description = description
+        tmp_db.session.commit()
+        plugin_id = (
+            tmp_db.session.query(RAMP.id)
+            .filter(RAMP.name == name, RAMP.version == version)
+            .first()[0]
+        )
+        return plugin_id
+    plugin = RAMP(
+        name=name,
+        description=description,
+        version=version,
+        tags=tags,
+    )
+    plugin.plugin_id = f"{name}@{version}"
+    tmp_db.session.add(plugin)
+    tmp_db.session.commit()
+    plugin_id = (
+        tmp_db.session.query(RAMP.id)
+        .filter(RAMP.name == name, RAMP.version == version)
+        .first()[0]
+    )
+    return plugin_id
+
+
+def create_plugins(
```

With property based testing you need to carefully consider what should be randomized and what should be fixed. Randomization creates more test cases (and can find more errors) but also limits the possible assertions you can make in the test case (which can hide errors). Something like this, that creates some plugins in the DB may be better as a known list of plugins implemented as a pytest fixture for some tests.